### ISSUE TITLE
Error creating instance: Cross-project references for subnetwork not allowed

### ISSUE THREAD

**Issue by anubbhavm** [hashicorp/terraform-provider-google#7368]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

* Terraform v0.12.20
* provider.google v3.40.0

### Affected Resource(s)

* google_compute_instance

### Terraform Configuration Files

```tf
module "billing_utilities_vm" {
 
 project_id   = module.billing_utilities_project.project_id
 machine_type = "n1-standard-1"
 zone         = "us-central1-a"
 purpose      = "connectivity"
 
 boot_disk = {
   initialize_params = {
     image = var.vm_image
   }
   mode = "READ_WRITE"
 }
 
 shielded_instance_config = {
   enable_secure_boot          = true
   enable_vtpm                 = true
   enable_integrity_monitoring = true
 }
 
network_interface = [{
 subnetwork = "https://www.googleapis.com/compute/v1/projects/cf-mps-cpl-abilling-n-3152/regions/us-central1/subnetworks/cf-aconnect-n-us-central1"
}]
 
 metadata = {
   enable-oslogin = "TRUE"
 }
 
 service_account = {
   email  = module.billing_utilities_serviceaccount.name
   scopes = ["cloud-platform"]
 }
}
```
### Actual Behavior

The VM is being created in the service project and the subnetwork is in the host project. The script fails with the following error:

```tf
module.billing_utilities_vm.google_compute_instance.standalone: Creating...

Error: Error creating instance: googleapi: Error 400: Invalid value for field 'resource.networkInterfaces[0]': '{  "subnetwork": "projects/cf-mps-cpl-abilling-n-3152/regions/us-central1/subnetworks/cf-aconnect-n-...'. Cross-project references for this resource are not allowed., invalid

  on .terraform/modules/billing_utilities_vm/main.tf line 21, in resource "google_compute_instance" "standalone":
  21: resource "google_compute_instance" "standalone" {
```

**Comment by edwardmedia** [comment]
@anubbhavm do you plan to use S[hared VPC](https://cloud.google.com/vpc/docs/shared-vpc)? If yes, have you enable [it ](https://www.terraform.io/docs/providers/google/r/compute_shared_vpc_host_project.html) ? If yes, can you post the other code and the debug log?

**Comment by anubbhavm** [comment]
@edwardmedia Yes, planning to use Shared VPC. However, I have enabled it manually through the console and not through Terraform. Attaching the debug log for your reference.
[terraform.log](https://github.com/hashicorp/terraform-provider-google/files/5293805/terraform.log)

**Comment by edwardmedia** [comment]
@anubbhavm can you post your log at https://gist.github.com/? Likely the host project is not visible to the service project.
Can you import the host and service projects and then post the state as well?

**Comment by anubbhavm** [comment]
@edwardmedia The SA running the TF script has `compute.networkUser` role on the subnet in the host project. Here's the gist: https://gist.github.com/anubbhavm/cb684150c3a572c4634e3ea2309990a4

**Comment by anubbhavm** [comment]
Figured out the issue, the host project was incorrectly specified. Thanks for your help!

**Comment by edwardmedia** [comment]
@anubbhavm I am glad you figured that out. To me, it was not obvious for anything wrong. Closing this issue then
